src/common.ts

```diff
@@ -100,3 +100,63 @@ export async function validateTxTip(node: GQLNodeInterface, ardriveOracle: ArDri
 
 	return tipPercentage + EPSILON >= 15 && wasValidTipRecipient;
 }
+
+export function daysDiffInEST(prev: Date, curr: Date): number {
+	const prevEstDate = dateToEST(prev);
+	const currEstDate = dateToEST(curr);
+
+	let daysCount = 0;
+	const cursorDate = new Date(prevEstDate.getTime());
+	cursorDate.setHours(0);
+	cursorDate.setMinutes(0);
+	cursorDate.setSeconds(0);
+
+	while (currEstDate.getTime() > cursorDate.getTime()) {
+		const cursorDay = cursorDate.getDate();
+		cursorDate.setDate(cursorDay + 1);
+		if (cursorDate.getTime() < currEstDate.getTime()) {
+			daysCount++;
+		}
+	}
+	return daysCount;
+}
+
+export function weeksDiffInEST(prev: Date, curr: Date): number {
+	const prevEstDate = dateToEST(prev);
+	const currEstDate = dateToEST(curr);
+
+	let weeksCount = 0;
+	const cursorDate = toSunday(prevEstDate);
+
+	while (currEstDate.getTime() > cursorDate.getTime()) {
+		const cursorDay = cursorDate.getDate();
+		const daysPerWeek = 7;
```

Comment:
sunday is 0 and saturday is 6

Reply:
What I do in the line below is to sum 7 days to the current one

Reply:
Ohhh you're right